js/utils/metrics.js

```diff
@@ -126,6 +126,11 @@ export function showMetricsModal(opts) {
 }
 
 export function recordEvent(key, segmentation) {
+  if (!key) throw new Error('Please provide a key');
+  if (segmentation && !_.isObject(segmentation)) {
```

Comment:
What does _.isObject give us that the native `typeof segmentation === 'object'` wouldn't?

Reply:
It detects null, typeof will say null is an object.

It's not too likely null would get passed in, but it seemed worth checking. I could also do typeof and !==null

Reply:
but in the first part you're checking for a `segmentation` truthy, which would be false if it was null.

Anyways, it's not a huge deal since we're using underscore anyway. In the web-buy, I imagine we'd be a bit more discriminating since page size is a bit bigger concern.

Reply:
Oh, duh. Your're right.